### PR TITLE
[CI] Upgrade intel support packages for XPU

### PR DESCRIPTION
**Pull request by mengfei25**
upgrade intel basekit package to 0.5 for XPU
Works for #114850

### CODE REVIEW

**Comment by mengfei25** [comment]
@chuanqi129 @EikanWang Please help have a review

**Review by EikanWang — APPROVED** [review]

File: .ci/docker/build.sh [hunk 1]
```diff
@@ -226,7 +226,7 @@ case "$image" in
     PROTOBUF=yes
     DB=yes
     VISION=yes
-    BASEKIT_VERSION=2024.0.0-49522
+    BASEKIT_VERSION=0.5
```
Comment (chuanqi129) [inline .ci/docker/build.sh:229]:
```suggestion
    XPU_VERSION=0.5
```

File: .ci/docker/common/install_xpu.sh [hunk 2]
```diff
@@ -40,11 +39,11 @@ function install_ubuntu() {
         mesa-vdpau-drivers mesa-vulkan-drivers va-driver-all vainfo hwinfo clinfo
     # Development Packages
     apt-get install -y libigc-dev intel-igc-cm libigdfcl-dev libigfxcmrt-dev level-zero-dev
-    # Install Intel® oneAPI Base Toolkit
+    # Install Intel Support Packages
     if [ -n "$BASEKIT_VERSION" ]; then
-        apt-get install intel-basekit=$BASEKIT_VERSION -y
+        apt-get install -y intel-for-pytorch-gpu-dev-${BASEKIT_VERSION}
```
Comment (chuanqi129) [inline .ci/docker/common/install_xpu.sh:44]: Same here, and apply to other places which used the `BASEKIT_VERSION`.

**Review by chuanqi129 — APPROVED** [review]

**Comment by mengfei25** [comment]
@atalman Could you please help to review? And looks like the current failures are not related with this PR change

**Review by atalman — APPROVED** [review]
LGTM, one cuda failure seems not to be related

**Pull request closed without merging**